File: work/prime/prime.rb
#!/usr/bin/env ruby
limit = 10000

prime_list = -> (num) {
  result = []
  list = Range.new(2, num).to_a
  while Math.sqrt(num) > list.first do
    current = list.shift
    result << current
    list = list.select { |v| (v % current) > 0 }
  end
  result + list
}

printf("%d\n", prime_list.call(limit).length)

File: work/prime/prime.js
limit = 10000

prime_list = function (num) {
    result = []
    list = Array.from(Array(num+1).keys())
    while (list[0] <= 1) { list.shift() }
    while (Math.sqrt(num) > list[0]) {
        current = list.shift()
        result.push(current)
        list = list.filter(function (v) { return (v % current) > 0 })
    }
    return result.concat(list)
}

console.log(prime_list(limit).length)
